treescript/mercurial.py

```diff
@@ -56,7 +56,14 @@ def build_hg_environment():
 
     """
     env = os.environ.copy()
-    env['HGRCPATH'] = HGRCPATH
+    if context.config['hgrc']:
+        env['HGRCPATH'] = context.config['hgrc']
+        if not os.path.exists(env['HGRCPATH']):
+            log.warning(FileNotFoundError(env['HGRCPATH']))
+            log.info('Using default hgrc')
+            env['HGRCPATH'] = HGRCPATH
+    else:
+        env['HGRCPATH'] = HGRCPATH
```

Comment:
I'm actually thinking we amend the default config ( https://github.com/mozilla-releng/treescript/blob/master/treescript/script.py#L63 ) 

Then we can assume context.config['hgrc'] exists and don't need to test it is set, we can still test for the file's existence, and maybe raise an exception instead of falling back here. (We'd raise the exception somewhere around https://github.com/mozilla-releng/treescript/blob/master/treescript/script.py#L41 maybe in a new helper function "check_config" before we create the aiohttp client session.